sorts/binary_search.cpp

```diff
@@ -0,0 +1,51 @@
+#include <fstream>
+#include <iostream>
+#include <vector>
+
+const int32_t binarySearch(const int32_t value, const std::vector<int32_t> vector, const size_t size) {
+    if (size < 0) {
+        throw std::invalid_argument("Given argument [size] is invalid - cannot be less than minimal treshold");
+    }
+
+    uint32_t midPoint = size / 2;
+    std::cout << "Starting at position [" << midPoint << "]\n";
+    
+    while (vector[midPoint] != value) {
+        const int32_t accessedValue = vector[midPoint];
+        std::cout << "Accessed value [" << accessedValue << "] at index [" << midPoint << "]\n";
+        if (value > accessedValue) {
+            midPoint = midPoint + ((size - midPoint) / 2);
+        }
+        else if (value < accessedValue) {
+            midPoint = midPoint - ((1 == midPoint) ? 1 : midPoint / 2);
+        }
+    }
+    
+    return midPoint;
+}
+
+int main() {
+    int32_t inputValuePosition = 0;
+    
+    try {
+        std::ifstream input("input.txt");
```

Comment:
Receive input path as cmd line parameter

Reply:
Should I use Boost library for that or use good old C-style cmd line argument parsing?

Reply:
For such an easy case you can just parse the options since all your arguments are positional